### ISSUE TITLE
Error when No Video Input Device Attached: TypeError: Cannot read properties of undefined (reading 'deviceId')

### ISSUE THREAD

**Issue by sahil2401** [zoom/videosdk-ui-toolkit-web#31]
I encountered an issue when trying to use the Zoom Video SDK UI Toolkit on a system that has no video input device attached. Specifically, when attempting to enable the "preview" feature, the following error occurs:

> `TypeError: Cannot read properties of undefined (reading 'deviceId')`

Steps to Reproduce:
Run the application on a system without any video input devices attached.
Attempt to join a session with the "preview" feature enabled.

Expected Behavior:
The application should gracefully handle the absence of video input devices without throwing an error, and the user should still be able to join the session.

Actual Behavior:
The application throws a TypeError, and the session join functionality fails.

Thank you for your attention to this issue!

![Screenshot 2024-09-18 at 4 45 12 PM](https://github.com/user-attachments/assets/0c98596c-6fc6-4068-9def-2d10bafddb8b)
![Screenshot 2024-09-18 at 4 45 27 PM](https://github.com/user-attachments/assets/653c9d8f-3886-4d61-88ef-d6a40d2b8c93)

**Comment by tommygaessler** [comment]
Hey @sahil2401 , thank you for reporting this bug. We will fix it in the next release. 

In the meantime, can you please share the use case for using a device with no video camera? It will help us understand and design better for this use case.

**Comment by sahil2401** [comment]
sure, 
I am using a Mac Mini, which doesn't have a built-in camera, so I use an external webcam for testing purposes. However, if no camera is attached, I encounter an error. It would be helpful to consider the use case where users might not have a video camera connected to their device. This can assist us in designing better solutions for such scenarios.

**Comment by tommygaessler** [comment]
Agree, we will include this fix in the next release, thank you for sharing the use case!

**Comment by k00ka** [comment]
We received an identical report from one of our users this week. They faced the same issue on our platform, which uses videosdk 2.1.5.

**Comment by tommygaessler** [comment]
Hey @k00ka , are you using the UI Toolkit or the Video SDK directly? The UI Toolkit is on 2.1.0 while the most recent version of the Video SDK is 2.1.5. 

https://developers.zoom.us/changelog/video-sdk/web/
https://developers.zoom.us/changelog/ui-toolkit/web/
